lib/sshkit/backends/abstract.rb

```diff
@@ -82,9 +82,10 @@ def execute(*args)
 
       def within(directory, &_block)
         (@pwd ||= []).push directory.to_s
+        escaped = pwd_path.shellescape.gsub("\\~", "~")
         execute <<-EOTEST, verbosity: Logger::DEBUG
-          if test ! -d #{File.join(@pwd).shellescape}
-            then echo "Directory does not exist '#{File.join(@pwd).shellescape}'" 1>&2
+          if test ! -d #{escaped}
+            then echo "Directory does not exist #{escaped}" 1>&2
```

Comment:
The removal of the single quotes is a very minor change but it does break a functional test.

Before:

```
Directory does not exist '/tmp'
```

After:

```
Directory does not exist /tmp
```

Would you mind restoring the quotes?